web_monitoring/pf_edgi.py

```diff
@@ -0,0 +1,172 @@
+    content = res.json()
+    assert content['status'] == 'ok'  # business logic is OK
+    assert content['result']['status'] == 'ok'
+    return content['result']
+
+
+def get_file(cabinet_id, archive_id, page_key):
+    uri = file_command_uri(cabinet_id, archive_id, page_key, 'file')
+    res = requests.get(uri)
+    assert res.ok  # server is OK
+    return res.content  # intentionally un-decoded bytes
+
+
+def format_version(*, url, dt, uri, version_hash, title, agency, site,
+                   metadata):
+    """
+    Format version info in preparation for submitting it to web-monitoring-db.
+
+    Parameters
+    ----------
+    url : string
+        page URL
+    dt : datetime.datetime
+        capture time
+    uri : string
+        URI of version
+    version_hash : string
+        sha256 hash of version content
+    title : string
+        primer metadata (likely to change in the future)
+    agency : string
+        primer metadata (likely to change in the future)
+    site : string
+        primer metadata (likely to change in the future)
+
+    Returns
+    -------
+    version : dict
+        properly formatted for as JSON blob for web-monitoring-db
+    """
+    # Existing documentation of import API is in this PR:
+    # https://github.com/edgi-govdata-archiving/web-monitoring-db/pull/32
+    return dict(
+         page_url=url,
+         page_title=title,
+         site_agency=agency,
+         site_name=site,
+         capture_time=dt.isoformat(),
+         uri=uri,
+         version_hash=version_hash,
+         source_type='page_freezer',
+         source_metadata=metadata
+    )
+
+
+def page_to_version(url, cabinet_id, archive_id, page_key, *,
+                    agency, site):
+    """
+    Obtain URI, timestamp, metadata, hash, and title and return a Version.
+    """
+    uri = file_command_uri(cabinet_id, archive_id, page_key, 'file')
+    dt = datetime.fromtimestamp(int(archive_id))
+    metadata = get_file_metadata(cabinet_id, archive_id, page_key)
+    content = get_file(cabinet_id, archive_id, page_key)
+    version_hash = hashlib.sha256(content).hexdigest()
+    title = utils.extract_title(content)
+    version = format_version(url=url, dt=dt, uri=uri,
+                             version_hash=version_hash, title=title,
+                             agency=agency, site=site,
+                             metadata=metadata)
+    return version
+
+
+def archive_to_versions(cabinet_id, archive_id, *, agency, site):
+    load_archive(cabinet_id, archive_id)
+    results = search_archive(cabinet_id, archive_id, '')['founds']
+    for result in results:
+        yield page_to_version(result['url'], cabinet_id, archive_id,
+                              result['key'], agency=agency, site=site)
+    unload_archive(cabinet_id, archive_id)
+
+
+def unique_urls(cabinets):
+    return set([entry['url']
+                for cabinet in cabinets.values()
+                for entry in cabinet])
+
+
+class Context:
```

Comment:
I usually try to avoid including bits that aren’t yet used in PRs, but I don’t really think it’s a big deal. No major reason to revise if you’re likely to start building on this pretty quickly.